osu.Game.Rulesets.Mania/Mods/ManiaModFadeIn.cs

```diff
@@ -19,5 +29,86 @@ public class ManiaModFadeIn : Mod
         public override double ScoreMultiplier => 1;
         public override bool Ranked => true;
         public override Type[] IncompatibleMods => new[] { typeof(ModFlashlight<ManiaHitObject>) };
+
+        private const float lanecover_size_filled = 0.5f;
+        private const float lanecover_size_gradient = 0.25f;
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<ManiaHitObject> drawableRuleset)
+        {
+            ManiaPlayfield maniaPlayfield = (ManiaPlayfield)drawableRuleset.Playfield;
+
+            foreach (Column column in maniaPlayfield.Stages.SelectMany(stage => stage.Columns))
+            {
+                column.hitObjectArea.ChildrenOfType<BufferedContainer>().First().Add(new LaneCover(false)
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    SizeFilled = lanecover_size_filled,
+                    SizeGradient = lanecover_size_gradient
+                });
+            }
+        }
+
+        private class LaneCover : CompositeDrawable
+        {
+            private readonly Box gradient;
+            private readonly Box filled;
+            private readonly bool reversed;
+
+            public LaneCover(bool reversed)
```

Comment:
this breaks if you change mania scrolling direction mid-gameplay. I assume this is what you intended the `reversed` ctor parameter for but it is only ever invoked with `false` as the value.

Reply:
You can bind the scrolling direction to the stage (maybe look at the implementation of keys for reference).

Reply:
Oh I completely forgot about up-scroll players. The ctor param was actually intended for re-use in HD, but I guess it can be used for that as well